shanoir-ng-users/src/main/java/org/shanoir/ng/user/service/VIPUserServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+ */
+@Component
+@ConditionalOnProperty(name = "vip.enabled", havingValue = "true")
+public class VIPUserServiceImpl implements VIPUserService{
+    /**
+     * Logger
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(VIPUserServiceImpl.class);
+
+    @Value("${vip.uri}")
+    private String vip_uri;
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private KeycloakServiceAccountUtils keycloakServiceAccountUtils;
+
+
+
+    @Override
+    public User createVIPAccountRequest(final User user) throws EntityNotFoundException, SecurityException, MicroServiceCommunicationException, PasswordPolicyException {
+
+        // Verify if the user is created in SHANOIR DB
+        final boolean userExist = userRepository.existsById(user.getId());
+        if (!userExist) {
+            LOG.error("User with id {} not found", user.getId());
+            throw new EntityNotFoundException(User.class, user.getId());
+        }
+
+        AccessTokenResponse accessTokenResponse = keycloakServiceAccountUtils.getServiceAccountAccessToken();
+
+        /* Password generation */
+        final String newPassword = PasswordUtils.generatePassword();
+        if (!PasswordUtils.checkPasswordPolicy(newPassword)) {
+            throw new PasswordPolicyException();
+        }
+
+        // vip user info;
+        String[] accountType = new String[] {"Support"};
+        String comments = "";
+        VIPUserLevel userLevel = VIPUserLevel.Beginner;
+        CountryCode countryCode = CountryCode.fr;
+        String institution = user.getAccountRequestInfo() == null ? "inria_admin_generated" : user.getAccountRequestInfo().getInstitution();
+
+        VIPUser vipUser = new VIPUser(user.getFirstName(), user.getLastName(), user.getEmail(), institution, newPassword, userLevel, countryCode, comments, accountType);
+
+        // prepare entity.
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessTokenResponse.getToken());
+        HttpEntity entity = new HttpEntity(vipUser, headers);
+
+        try {
+            restTemplate.exchange(this.vip_uri, HttpMethod.POST, entity, Void.class);
+            return user;
+        }catch (HttpStatusCodeException e) {
+            // in case of an error with response payload
+            LOG.error("error while saving vip user with status : {} ,and message : {}", e.getStatusCode(), e.getMessage());
+            throw new MicroServiceCommunicationException("Error while communicating with VIP with status : "+e.getStatusCode()+ ",and message : "+e.getMessage());
+        } catch (RestClientException e) {
+            // in case of an error with no response payload
+            LOG.error("there is no response payload while saving vip user");
+            throw new MicroServiceCommunicationException("Error while communicating with VIP with no response payload while saving vip user");
```

Comment:
I suggest passing the parent exception as a parameter : 
``` java
} catch (RestClientException e) {
   // in case of an error with no response payload
   LOG.error("there is no response payload while saving vip user", e);
   throw new MicroServiceCommunicationException("Error while communicating with VIP with no response payload while saving vip user", e);
```